crates/runtime/src/http/routes.rs

```diff
@@ -242,13 +242,28 @@ pub(crate) fn routes(
     unauthenticated_router
         .merge(authenticated_router)
         .route_layer(middleware::from_fn_with_state(rt.status(), check_shutdown))
+        .route_layer(middleware::from_fn(with_datafusion_context))
         .route_layer(middleware::from_fn(track_metrics))
         .layer(Extension(Arc::clone(&rt.app)))
+        .layer(Extension(Arc::clone(&rt.df)))
         .layer(cors_layer(cors_config))
 }
 
+async fn with_datafusion_context(mut req: Request<Body>, next: Next) -> impl IntoResponse {
```

Comment:
I don't see a benefit to adding this to the `req` extensions since its already added to the request context.

Reply:
That would give us 3 ways to get access to the DataFusion context, which is confusing. We should just have one way.

Reply:
removed